pkg/executor/sortexec/topn.go

```diff
@@ -17,147 +17,232 @@ package sortexec
+//                   │ Chunk Fetcher │                                                                        
+//                   └───────┬───────┘                                                                        
+//                           │                                                                                
+//                           │                                                                                
+//                           ▼                                                                                
+//                      Check Spill──────►Spill Triggered─────────►Spill                                      
+//                           │                                       │                                        
+//                           ▼                                       │                                        
+//                   Spill Not Triggered                             │                                        
+//                           │                                       │                                        
+//                           ▼                                       │                                        
+//                         Push◄─────────────────────────────────────┘                                        
+//                           │                                                                                
+//                           ▼                                                                                
+//      ┌────────────────►Channel◄───────────────────┐                                                        
+//      │                    ▲                       │                                                       ▼
+//      │                    │                       │                                                        
+//    Fetch                Fetch                   Fetch                                                      
+//      │                    │                       │                                                        
+// ┌────┴───┐            ┌───┴────┐              ┌───┴────┐                                                   
+// │ Worker │            │ Worker │   ......     │ Worker │                                                   
+// └────┬───┘            └───┬────┘              └───┬────┘                                                   
+//      │                    │                       │                                                        
+//      │                    │                       │                                                        
+//      │                    ▼                       │                                                        
+//      └───────────► Multi-way Merge◄───────────────┘                                                        
```

Comment:
Is there still multiple worker fetch when spill does not triggered?

Reply:
This graph looks like multiple workers fetch regardless of whether it is spilled or not

Reply:
> This graph looks like multiple workers fetch regardless of whether it is spilled or not

> Is there still multiple worker fetch when spill does not triggered?

No, there is no worker when spill is not triggered and L161 explains that the picture shows the the procedure of topn when spill is triggered.

Reply:
ok